### PR TITLE
Flickrdata

### PR DESCRIPTION
**Pull request by unamaria**
@sharnik 
I created a FlickrDetail model (first it was data but I had problems with pluralize datum-data) that stores the username and flickr id. When a user adds a flickr username (there's a link in profile) the controller calls the model method to get the flickr id and saves it. What do you think?

### CODE REVIEW

File: app/models/flickr_detail.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		@api_key = 'b8f714a84f67af6df3b0901e4a3ce803'
+		@root = 'https://api.flickr.com'
+		@api_path = '/services/rest/?'
+		@format = 'json'
+		@conn = Faraday.new(:url => @root) do |faraday|
+		  faraday.request  :url_encoded             # form-encode POST params
+		  faraday.response :logger                  # log requests to STDOUT
+		  faraday.adapter  Faraday.default_adapter  # make requests with Net::HTTP
+		end
+	end
```
Comment (sharnik) [inline app/models/flickr_detail.rb]: I would move those two methods to FlickrContentService, as they are about external API, not your internal model/database persistence.

**Comment by unamaria** [comment]
@sharnik 
OK! Done, I searched how to access the service methods from the controller and it works, should it be like that?
Now I'd like to create BlogItem and FlickrItem instances, is it a good idea to test it from the same service (create_flickr_items method) until I don't have the class that will take care of fetching all new data?

**Comment by sharnik** [comment]
I'm not exactly sure what you mean with "is it a good idea to test it from the same service (create_flickr_items method) until I don't have the class that will take care of fetching all new data", but you should try and I'll tell you if it was a good idea ;)

**Comment by unamaria** [comment]
Hehe I did and it's kind of working, now there's a method that calls the requests methods and creates a BlogItem and a FlickrItem instance per photo_id, I guess that should stay there and this method would be called in the chroned task. I'll keep working on this branch!

**Comment by unamaria** [comment]
@sharnik 
I polished details and now when calling create_flickr_items method (in FlickrContentService) everything works fine, the model has direct_url and flickr_url methods and photos are rendered correctly in the main view (blog items index). But I'm making a query in the flickr_items partial to get the flickr id because I don't have a controller corresponding to that view, should I create one?

Now I'm not sure if keep working on this (todo: get only new content, implement galleries) or let it rest and start with something else. In any case, I'd like to merge this first :)

What do you think? Thanks!

**Comment by sharnik** [comment]
Merge it, let it settle. ou can get back to it in after a short break.

File: app/views/flickr_items/_flickr_item.html.erb [hunk 2]
```diff
@@ -1,5 +1,8 @@
 <h2><%= flickr_item.title %></h2>
 
-<a data-flickr-embed="true" href="https://www.flickr.com/photos/46626331@N07/<%= flickr_item.photo_id %>"><img src="https://farm9.staticflickr.com/8707/<%= flickr_item.photo_id %>_5908ab7d1d_k.jpg" alt=<%= flickr_item.title %>></a>
+<a data-flickr-embed="true" href="<%= flickr_item.flickr_url(FlickrDetail.find_by_user_id(flickr_item.blog_item.user_id).nsid, flickr_item.photo_id) %>">
```
Comment (sharnik) [inline app/views/flickr_items/_flickr_item.html.erb:3]: You need some method for that one: "FlickrDetail.find_by_user_id(flickr_item.blog_item.user_id).nsid"

Maybe define it as flickr_item#owner_nsid ?

**Comment by unamaria** [comment]
done! (commit a4d2ad596f447e30659ccadffa57d0cbd075955d)

File: app/views/flickr_items/_flickr_item.html.erb (hunk 2 above)
Comment (unamaria) [inline app/views/flickr_items/_flickr_item.html.erb:3]: done! (commit a4d2ad596f447e30659ccadffa57d0cbd075955d)

**Pull request merged**